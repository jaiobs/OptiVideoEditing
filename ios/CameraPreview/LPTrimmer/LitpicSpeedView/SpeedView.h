//
//  SpeedView.h
//  react-native-litpic-camera-module
//
//  Created by optisol on 03/08/20.
//

#import <UIKit/UIKit.h>
#import <AVFoundation/AVFoundation.h>
#import <React/RCTViewManager.h>
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>



@class SpeedView;


NS_ASSUME_NONNULL_BEGIN

@interface SpeedView : RCTViewManager <RCTBridgeModule>
- (void)sendData;
@end

NS_ASSUME_NONNULL_END

