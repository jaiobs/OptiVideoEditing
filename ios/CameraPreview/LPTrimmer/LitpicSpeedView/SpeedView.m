//
//  SpeedView.m
//  react-native-litpic-camera-module
//
//  Created by optisol on 03/08/20.
//

#import "SpeedView.h"
#import "RNFileSystem.h"
#import "RNImageUtils.h"
#import <React/RCTBridge.h>
#import <React/RCTUIManager.h>
#import <React/RCTEventDispatcher.h>
#import <React/RCTLog.h>
#import <React/RCTUtils.h>
#import <React/UIView+React.h>
#import "react_native_litpic_camera_module-Swift.h"
#import "RNCamera.h"
#import "RNSensorOrientationChecker.h"
#import "CommonEmittor.h"


@implementation SpeedView

RCT_EXPORT_MODULE(SpeedView);

SpeedViewSwift * SVS;
RNSensorOrientationChecker * sensorOrientationCheckerK;


- (UIView *)view {
    SVS = [[SpeedViewSwift alloc] init];
    sensorOrientationCheckerK = [RNSensorOrientationChecker new];
    [[NSNotificationCenter defaultCenter] addObserver:self
                                             selector:@selector(orientationChanged:)
                                                 name:UIApplicationDidChangeStatusBarOrientationNotification
                                               object:nil];
    
    [SVS copyEmitorWithCompletionHandler:^(NSDictionary *response) {
        [self sendData:response];
    }];

    return SVS;
}



RCT_CUSTOM_VIEW_PROPERTY(videos, NSArray, SpeedViewSwift){
    [SVS setVideoUrl:[RCTConvert NSArray:json]];
}

RCT_CUSTOM_VIEW_PROPERTY(audio, NSDictionary, SpeedViewSwift){
    [SVS setAudioInfo:[RCTConvert NSDictionary:json]];
}



- (void)sendData:(NSDictionary*)response{
    CommonEmittor *events = [[CommonEmittor alloc] init];
    [events setBridge:self.bridge];
    [events sendCommonEvent:response];
}


RCT_EXPORT_METHOD(refreshData){
    dispatch_async(dispatch_get_main_queue(), ^{
        [SVS reloadData];
    });
}


RCT_EXPORT_METHOD(updateSpeed:(NSInteger)level callback:(RCTResponseSenderBlock)callback){
    [SVS updateSpeedModeWithSpeedLevel:level completionHandler:^(NSDictionary *response){
        callback(@[response]);
    }];
}

RCT_EXPORT_METHOD(getVideo:(double)time callback:(RCTResponseSenderBlock)callback){
    dispatch_async(dispatch_get_main_queue(), ^{
        [SVS applySpeedModeWithCompletionHandler:^(NSDictionary *response) {
                    callback(@[response]);
        }];
    });
}

RCT_EXPORT_METHOD(getVideoTime:(RCTResponseSenderBlock)callback){
    dispatch_async(dispatch_get_main_queue(), ^{
        [SVS checkVideoTimeWithCompletionHandler:^(NSDictionary *response) {
            callback(@[response]);
        }];
    });
}



- (void)orientationChanged:(NSNotification *)notification{
    UIInterfaceOrientation orientation = [UIApplication sharedApplication].statusBarOrientation;
    if (UIInterfaceOrientationIsLandscape(orientation)) {
        if (SVS != nil){
//            [SVS orientaionChangedWithIsPortrait:FALSE];
        }
        NSLog(@"Landscape");
    }else if (UIInterfaceOrientationIsPortrait(orientation)){
        NSLog(@"Portrait");
        if (SVS != nil){
//            [SVS orientaionChangedWithIsPortrait:TRUE];
        }
    }else{
        
    }
    
}

@end




