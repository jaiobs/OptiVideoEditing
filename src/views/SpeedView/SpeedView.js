import React, { Component } from "react";
import { View, StyleSheet, Text, TouchableOpacity, Image, NativeModules, Alert, SafeAreaView, NativeEventEmitter } from "react-native";
import { requireNativeComponent } from 'react-native';

import { Loader } from "../../components/ViewComponents/Loader";
const VideoSpeedView = requireNativeComponent('SpeedView', null);
var VSVActions = NativeModules.SpeedView;
import { Images, Colors } from "../../res";

export default class SpeedView extends Component {
  constructor(props) {
    super(props);
    this.state = {
      audio: this.props.audioInfo,
      IsSingleVideoSelected: false,
      videoPath: this.props.videoPath,
      IsSlowSpeedTwo: false,
      IsSlowSpeedOne: false,
      IsFastSpeedTwo: false,
      IsFastSpeedOne: false,
      isVideoProcessing: false
    }
  }

  componentDidMount() {
    const myModuleEvt = new NativeEventEmitter(NativeModules.CommonEmittor)
    myModuleEvt.addListener('CommonEmittorEvent', (response) => {
      console.log(response);
      if (response.action == "videoSelected") {
        if (response.speed == 5) {
          this.setState({
            IsSlowSpeedTwo: false,
            IsSlowSpeedOne: false,
            IsFastSpeedTwo: true,
            IsFastSpeedOne: false,
          })
        } else if (response.speed == 4) {
          this.setState({
            IsSlowSpeedTwo: false,
            IsSlowSpeedOne: false,
            IsFastSpeedTwo: false,
            IsFastSpeedOne: true,
          })
        } else if (response.speed == 3) {
          this.setState({
            IsSlowSpeedTwo: false,
            IsSlowSpeedOne: false,
            IsFastSpeedTwo: false,
            IsFastSpeedOne: false,
          })
        } else if (response.speed == 2) {
          this.setState({
            IsSlowSpeedTwo: false,
            IsSlowSpeedOne: true,
            IsFastSpeedTwo: false,
            IsFastSpeedOne: false,
          })
        } else if (response.speed == 1) {
          this.setState({
            IsSlowSpeedTwo: true,
            IsSlowSpeedOne: false,
            IsFastSpeedTwo: false,
            IsFastSpeedOne: false,
          })
        } else {
          this.setState({
            IsSlowSpeedTwo: false,
            IsSlowSpeedOne: false,
            IsFastSpeedTwo: false,
            IsFastSpeedOne: false,
          })
        }
      }
    }
    )
  }


  videoProcessing = () => {
    let value = VSVActions.getVideoTime((responseData) => {

      this.setState({
        isVideoProcessing: false
      })

      if (responseData.video_limit_exit == true) {
        Alert.alert(
          "Warning",
          "Video time greater then 30 seconds please reduce the size",
          { cancelable: true }
        );
      } else {
        this.props.onClose();
      }
    });
  }
  setVideoSpeedLevel = (speed) => {
    if (this.props.hasGif == true){
      let value = VSVActions.updateSpeed(speed, (response) => {
        this.timeoutHandle = setTimeout(() => {
        this.props.sticker.getMediaResponse((resp) => {
          this.timeoutHandle = setTimeout(() => {
                VSVActions.refreshData();
                this.setState({
                  isVideoProcessing: false
                })
              }, 1000);
        });
        }, 1000);

      })
    }else{
      let value = VSVActions.updateSpeed(speed, (response) => {
        this.timeoutHandle = setTimeout(() => {
          VSVActions.refreshData();
          this.setState({
            isVideoProcessing: false
          })
        }, 1000);
      })
    }
  }


  render() {
    return (
      <SafeAreaView style={{ flex: 1, backgroundColor:'black' }}>

        <VideoSpeedView videos={this.state.videoPath} audio={this.state.audio} style={{ flex: 1 }}></VideoSpeedView>

        <TouchableOpacity
          style={[styles.closeBtnContainer, { paddingLeft: this.props.orientationCheck == "portrait" ? 4 : 10 }]}
          onPress={() => {
            this.props.onClose();
          }}
        >
          <Image style={styles.close_icon} source={Images.close_icon} />
        </TouchableOpacity>

        <View style={{
          position: 'absolute', bottom: 125, width: '100%', height: 50, flexDirection: 'row',
          justifyContent: 'space-evenly', alignItems: 'center', alignContent: 'center'
        }}>
          <TouchableOpacity style={[styles.optionButton, { backgroundColor: this.state.IsSlowSpeedTwo ? Colors.primaryAccent : "white" }]} onPress={() => {
            this.setState({
              IsSlowSpeedTwo: true,
              IsSlowSpeedOne: false,
              IsFastSpeedOne: false,
              IsFastSpeedTwo: false,
              isVideoProcessing: true
            }, () => { this.setVideoSpeedLevel(5)});
          }}><Text style={{ color: this.state.IsSlowSpeedTwo ? 'white' : 'black', fontSize: 12 }}>{"3x slow"}</Text></TouchableOpacity>
          <TouchableOpacity style={[styles.optionButton, { backgroundColor: this.state.IsSlowSpeedOne ? Colors.primaryAccent : "white" }]} onPress={() => {
            this.setState({
              IsSlowSpeedTwo: false,
              IsSlowSpeedOne: true,
              IsFastSpeedOne: false,
              IsFastSpeedTwo: false,
              isVideoProcessing: true

            }, () => { this.setVideoSpeedLevel(4)});
          }}><Text style={{ color: this.state.IsSlowSpeedOne ? 'white' : 'black', fontSize: 12 }}>{"2x slow"}</Text></TouchableOpacity>
          <TouchableOpacity style={{
            width: 50,
            height: 50,
            justifyContent: 'center',
            alignContent: 'center',
            alignItems: 'center',
          }} onPress={() => {
            this.setState({
              IsSlowSpeedTwo: false,
              IsSlowSpeedOne: false,
              IsFastSpeedOne: false,
              IsFastSpeedTwo: false,
              isVideoProcessing: true
            }, () => { this.setVideoSpeedLevel(3)});
          }}>
            <Image style={[styles.timerIcon]} source={Images.slowmotion_icon} />
          </TouchableOpacity>
          <TouchableOpacity style={[styles.optionButton, { backgroundColor: this.state.IsFastSpeedOne ? Colors.primaryAccent : "white" }]} onPress={() => {
            this.setState({
              IsSlowSpeedTwo: false,
              IsSlowSpeedOne: false,
              IsFastSpeedOne: true,
              IsFastSpeedTwo: false,
              isVideoProcessing: true
            }, () => { this.setVideoSpeedLevel(2)});

          }}><Text style={{ color: this.state.IsFastSpeedOne ? 'white' : 'black', fontSize: 12 }}>{"2x fast"}</Text></TouchableOpacity>
          <TouchableOpacity style={[styles.optionButton, { backgroundColor: this.state.IsFastSpeedTwo ? Colors.primaryAccent : "white" }]} onPress={() => {
            this.setState({
              IsSlowSpeedTwo: false,
              IsSlowSpeedOne: false,
              IsFastSpeedOne: false,
              IsFastSpeedTwo: true,
              isVideoProcessing: true
            }, () => { this.setVideoSpeedLevel(1)});

          }}><Text style={{ color: this.state.IsFastSpeedTwo ? 'white' : 'black', fontSize: 12 }}>{"3x fast"}</Text></TouchableOpacity>
        </View>

      
      
        <View style={{
          position: 'absolute', bottom: 20, right: 0, width: 50, height: 50, maxHeight: 50, flexDirection: 'row',
          justifyContent: 'flex-end', alignItems: 'center', alignContent: 'center'
        }}>
          <TouchableOpacity style={[{ right: 10, width: 30, height: 30, justifyContent: 'center', alignItems: 'center' }]} onPress={() => this.videoProcessing()} >
            <Image style={[styles.imageIcon]} source={Images.next_video_icon} />
          </TouchableOpacity>
        </View>
       
       
        <Loader visibility={this.state.isVideoProcessing} />
      </SafeAreaView>
    )
  }
}


const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.black,
  },
  textButtons: {
    height: 40,
    width: 40,
    marginTop: 5,
    alignItems: 'center',
    justifyContent: 'center',
  },
  optionButton: {
    width: 65,
    height: 28,
    justifyContent: 'center',
    alignContent: 'center',
    alignItems: 'center',
    borderRadius: 14,
    backgroundColor: 'white'
  },
  textImage: {
    height: 30,
    width: 30,
    alignSelf: "center",
    resizeMode: "contain"
  },
  playerStyle: {
    flex: 1,
    width: "98%",
    height: "100%",
    alignSelf: "center",
    marginLeft: 8,
    marginRight: 8
  },
  headerContainer: {
    position: 'absolute',
    margin: 10,
    width: 50,
    height: 50,
  },
  footerContainer: {},
  closeBtnContainer: {
    padding: 4,
    marginTop:34,
    position: 'absolute'
  },
  nextBtnContainer: {
    alignSelf: "flex-end",
    margin: 10
  },
  video_next: {
    height: 24,
    width: 24
  },
  actionContainerRight: {
    position: "absolute",
    right: 0,
    top: 0,
    paddingTop: 10,
    paddingLeft: 15,
    width: 80,
    zIndex: 999,
    height: "100%",
    backgroundColor: "black",
    flexDirection: "column",
    justifyContent: "space-between"
  },
  actionContainerLeft: {
    position: "absolute",
    bottom: 20,
    paddingLeft: 10,
    paddingRight: 10,
    flexDirection: "row",
    justifyContent: "space-between"
  },
  actionContainerBottom: {
    position: "absolute",
    width: "100%",
    bottom: 20,
    paddingLeft: 10,
    paddingRight: 10,
    flexDirection: "row",
    justifyContent: "space-between"
  },
  close_icon: {
    height: 25, width: 25, margin: 10, resizeMode: "contain"
  },
  imageIcon: { height: 25, width: 25, margin: 10, resizeMode: "contain" },
  timerIcon: { height: 35, width: 35, margin: 10, resizeMode: "contain" }
});
