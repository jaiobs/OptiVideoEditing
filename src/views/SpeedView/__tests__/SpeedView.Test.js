import React from 'react';
import renderer from 'react-test-renderer';
import SpeedView from '../SpeedView'
import {
  NativeModules,
  NativeEventEmitter
} from "react-native";
import { shallow } from 'enzyme';
import Enzyme from 'enzyme';

describe("SpeedView-Test", () => {

  beforeEach(() => {
    SpeedView.componentDidMount = jest.fn();
  });

  NativeModules.SpeedView = {
    updateSpeed: (params) => {
      return Promise.resolve({
        IsSpeedUpdated: true 
      });
    }
  }
    
  it("should render the SpeedView", async () => {
    const tree = renderer
      .create(<SpeedView/>)
      .toJSON();
    ;
    expect(tree).toMatchSnapshot();
  })
})