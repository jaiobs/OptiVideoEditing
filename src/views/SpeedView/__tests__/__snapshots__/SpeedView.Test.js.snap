// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpeedView-Test should render the SpeedView 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "black",
      "flex": 1,
    }
  }
>
  <SpeedView
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginTop": 34,
        "opacity": 1,
        "padding": 4,
        "paddingLeft": 10,
        "position": "absolute",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/res/assets/images/cross.png",
        }
      }
      style={
        Object {
          "height": 25,
          "margin": 10,
          "resizeMode": "contain",
          "width": 25,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "bottom": 125,
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "space-evenly",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 14,
          "height": 28,
          "justifyContent": "center",
          "opacity": 1,
          "width": 65,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 12,
          }
        }
      >
        3x slow
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 14,
          "height": 28,
          "justifyContent": "center",
          "opacity": 1,
          "width": 65,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 12,
          }
        }
      >
        2x slow
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "width": 50,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/res/assets/images/slowmotion_icon.png",
          }
        }
        style={
          Array [
            Object {
              "height": 35,
              "margin": 10,
              "resizeMode": "contain",
              "width": 35,
            },
          ]
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 14,
          "height": 28,
          "justifyContent": "center",
          "opacity": 1,
          "width": 65,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 12,
          }
        }
      >
        2x fast
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 14,
          "height": 28,
          "justifyContent": "center",
          "opacity": 1,
          "width": 65,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 12,
          }
        }
      >
        3x fast
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "bottom": 20,
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "flex-end",
        "maxHeight": 50,
        "position": "absolute",
        "right": 0,
        "width": 50,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": 30,
          "justifyContent": "center",
          "opacity": 1,
          "right": 10,
          "width": 30,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/res/assets/images/next_button_send.png",
          }
        }
        style={
          Array [
            Object {
              "height": 25,
              "margin": 10,
              "resizeMode": "contain",
              "width": 25,
            },
          ]
        }
      />
    </View>
  </View>
</RCTSafeAreaView>
`;
